message Request {
    required string url = 1;
    optional string user_agent = 2;
    optional string content_type_limit = 3;
    optional int32 fetch_timeout = 4;
    optional int32 connect_timeout = 5;
    optional bool parse_stream = 6;
    optional bool record_stream = 7;
}

message Task {
    required int64 id = 1;
    optional Request request = 2;
    optional string consumer = 3;
}

message TaskResult {
    enum Type {
        RESPONSE = 1;
        ERROR = 2;
        STREAM_META = 3;
        STRIPE_COMMIT = 4;
    }

    required int64 task_id = 1;
    required Type type = 2;
    required int64 timestamp = 3;

    optional int32 error = 4;
    optional string meta = 5;

    optional int32 status_code = 6;
    optional bytes content = 7;
    repeated string headers = 8;

    optional int32 stripe_id = 9;
    optional int64 stripe_offset = 10;
    optional int32 stripe_written = 11;

    optional bool compressed = 16;
    optional string error_text = 17;
}

message BrokerResponse {
    enum Status {
        OK = 1;
        TASK = 2;
        TASK_GONE = 3;
    }
    required Status status = 1;
    optional Task task = 2;
}
